MoleculeVAE-reImplementation/dataloader.py

```diff
@@ -1,133 +1,272 @@
+    If the molecule cannot be calculated by RDkit, returns error
+    These values calculated here sometimes disagree with the values given in the dataset from the original
+    dataset.
+    """
+    m = Chem.MolFromSmiles(smiles)
+    logp = Chem.Descriptors.MolLogP(m)
+    qed = QED.default(m)
+    sas = sascorer.calculateScore(m)
 
+    return logp, qed, sas
 
-if __name__ == '__main__':
-    parser = argparse.ArgumentParser()
-    parser.add_argument('-e', '--exp_file',
-                        help='experiment file', default='expParam.json')
-    parser.add_argument('-d', '--directory',
-                        help='exp directory', default=None)
-    args = vars(parser.parse_args())
-    if args['directory'] is not None:
-        args['exp_file'] = os.path.join(args['directory'], args['exp_file'])
 
-    params = hyperparameters.load_params(args['exp_file'])
+def extract_properties_value(smile_onehot, smile_string):
+    """Extract properties value for latent space prediction
+    - Inputs:
+        - smile_onehot: one hot vector for training.
+        - smile_string: corresponding smile string.
+    - Returns:
+        - X_final: one hot vector which has the corresponding properties value
+        - Y_values:properties value
+    """
+    X_final = []
+    Y_values = []
+    label = []
+    if os.path.exists("properties_value.json"):
+        with open("properties_value.json", "r", encoding="utf-8") as json_file:
+            properties_dic = ast.literal_eval(json.load(json_file))
 
+            for x, s_string in zip(smile_onehot, smile_string):
+                try:
+                    y = properties_dic[str(s_string)]
+                except:
+                    y = caculate_target_values(s_string)
+                    temp_result = {str(s_string): y}
+                    properties_dic.update(temp_result)
+                X_final.append(x)
+                Y_values.append(y)
+                label.append(s_string)
+    else:
+        for x, s_string in zip(smile_onehot, smile_string):
+            try:
+                y = caculate_target_values(s_string)
 
+            except:
+                continue
+            X_final.append(x)
+            Y_values.append(y)
+            label.append(s_string)
+            properties_dic = dict(zip(label, Y_values))
+    """creat & update properties_value.json"""
+    with open("properties_value.json", "w") as outfile:
+        json.dump(str(properties_dic), outfile)
+    return X_final, Y_values
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-e", "--exp_file", help="experiment file", default="expParam.json"
+    )
+    parser.add_argument("-d", "--directory", help="exp directory", default=None)
+    args = vars(parser.parse_args())
+    if args["directory"] is not None:
+        args["exp_file"] = os.path.join(args["directory"], args["exp_file"])
 
+    params = hyperparameters.load_params(args["exp_file"])
```

Comment:
It seems like the `__main__` function here just loads the parameters but doesn't do anything else. Is that as you intended? If it's just for checking it might be good to add a comment to say so.